source-code/ide-extension/src/main.ts

```diff
@@ -27,20 +31,15 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
 			properties: {
 				vscode_version: vscode.version,
 				version: version,
-				workspaceRecommendation: !(await isDisabledRecommendation()),
-				autoConfigFileCreation: !(await isDisabledConfigFileCreation()),
+				$set: {
+					// this information are user based, it will help us to create cohorts and to get a understanding why some projects are exclude
+					"user-IDE-settings": {
+						"workspace-recommendation-setting": !(await isDisabledRecommendation()),
```

Comment:
use boolean property names here. 

`workspaceRecommendationSettingIsEnabled`: true | false  
`autoConfigRecommendationSettingIsEnabled`: true | false  

**Why?**

- the `is` indicates that the property is a boolean and thereby increases comprehension 
- i was confused why you negate the values with `!` until i realized that the properties should be interpreted as  "default truthy". this is ambiguous. You can take the ambiguity out by stating the default case `isEnabled`